README.md

```diff
@@ -29,7 +29,7 @@ fn main() -> Result<(), TrieError> {
     let value = b"test-value";
 
     let root = {
-        let mut trie = EthTrie::new(Arc::clone(&memdb));
+        let mut trie = EthTrie::new(memdb.clone());
```

Comment:
*nitpick* personally, I lean towards explicit `Arc` clones - just so it's obvious that you're cloning an `Arc`, but this is not a strong opinion

Reply:
I'm not sure I've landed on a global preference. You might be right generally. But at least in this README, the initialization as an `Arc` is only a few lines up, so my intuition is that the shorter line is better.